cdap-cli/src/main/java/co/cask/cdap/cli/command/schedule/AddTimeScheduleCommand.java

```diff
@@ -0,0 +1,109 @@
+    String programType = arguments.get(ArgumentName.PROGRAM_TYPE.toString()).toUpperCase();
+    String[] programIdParts = arguments.get(ArgumentName.PROGRAM.toString()).split("\\.");
+    String scheduleDescription = arguments.getOptional(ArgumentName.DESCRIPTION.toString(), "");
+    String cronExpression = arguments.get(ArgumentName.CRON_EXPRESSION.toString());
+    String schedulePropertiesString = arguments.getOptional(ArgumentName.SCHEDULE_PROPERTIES.toString(), "");
+    String scheduleRunConstraintsString = arguments.getOptional(ArgumentName.SCHEDULE_RUN_CONSTRAINTS.toString(), "");
+
+    if (scheduleIdParts.length < 2 || programIdParts.length < 2 || !scheduleIdParts[0].equals(programIdParts[0])) {
+      throw new CommandInputError(this);
+    }
+
+    String appId = scheduleIdParts[0];
+    String scheduleName = scheduleIdParts[1];
+    ScheduleId scheduleId = cliConfig.getCurrentNamespace().app(appId).schedule(scheduleName);
+
+    Map<String, String> constraintsMap =
+      ArgumentParser.parseMap(scheduleRunConstraintsString, ArgumentName.SCHEDULE_RUN_CONSTRAINTS.toString());
+
+    Schedules.Builder builder = Schedules.builder(scheduleName);
+    if (constraintsMap.containsKey("maxConcurrentRuns")) {
+      builder.setMaxConcurrentRuns(Integer.valueOf("maxConcurrentRuns"));
+    }
+    if (scheduleDescription != null) {
+      builder.setDescription(scheduleDescription);
+    }
+    Schedule schedule = builder.createTimeSchedule(cronExpression);
+
+    Map<String, String> programMap = ImmutableMap.of("programName", programIdParts[1],
+                                                     "programType", programType);
+    Map<String, String> propertiesMap = ArgumentParser.parseMap(schedulePropertiesString,
+                                                                ArgumentName.SCHEDULE_PROPERTIES.toString());
+
+    ScheduleInstanceConfiguration configuration =
+      new ScheduleInstanceConfiguration("TIME", schedule, programMap, propertiesMap);
+
+    scheduleClient.add(scheduleId, configuration);
+    printStream.printf("Successfully added schedule '%s' in program '%s'\n", scheduleName, appId);
+  }
+
+  @Override
+  public String getPattern() {
+    return String.format("add time schedule <%s> for <%s> <%s> " +
+                           "[description <%s>] at <%s> [properties <%s>] [constraints <%s>]",
+                         ArgumentName.SCHEDULE, ArgumentName.PROGRAM_TYPE, ArgumentName.PROGRAM,
+                         ArgumentName.DESCRIPTION, ArgumentName.CRON_EXPRESSION, ArgumentName.SCHEDULE_PROPERTIES,
+                         ArgumentName.SCHEDULE_RUN_CONSTRAINTS);
+  }
+
+  @Override
+  public String getDescription() {
+    return String.format("Adds %s which is associated with %s given.",
+                         Fragment.of(Article.A, ElementType.SCHEDULE.getName()),
+                         Fragment.of(Article.THE, ElementType.PROGRAM.getName()));
```

Comment:
Actually, I think the `A` should be blue too but I guess their regex doesn't match single-letter all-caps variables.